From 8be2417f7f9e688d3e383abc2109121cc691960e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Thu, 11 Sep 2025 12:14:17 +0200
Subject: [PATCH 081/553] cmake: Don't create framework header directories
 unless we have headers

qt_copy_framework_headers was unconditionally creating all the private,
qpa, rhi, and ssg header subdirectories, even if we didn't have any
headers to put in them, which doesn't match how we structure the
normal include directories.

Pick-to: 6.9 6.8
Change-Id: I9efe2fd9c4f6d10f57e30ffcba2b912dae35a2c4
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit c4f8fa83db2b3c0d1d2c56904c5f977e147725ae)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 cmake/QtFrameworkHelpers.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/QtFrameworkHelpers.cmake b/cmake/QtFrameworkHelpers.cmake
index 1982937aaf8..b48e39ccd8b 100644
--- a/cmake/QtFrameworkHelpers.cmake
+++ b/cmake/QtFrameworkHelpers.cmake
@@ -96,7 +96,6 @@ function(qt_copy_framework_headers target)
     foreach(type IN ITEMS PUBLIC PRIVATE QPA RHI SSG)
         set(in_files_${type} "")
         set(fw_output_header_dir "${output_dir_${type}}")
-        list(APPEND out_dirs "${fw_output_header_dir}")
         foreach(hdr IN LISTS arg_${type})
             get_filename_component(in_file_path ${hdr} ABSOLUTE)
             get_filename_component(in_file_name ${hdr} NAME)
@@ -105,6 +104,7 @@ function(qt_copy_framework_headers target)
             list(APPEND in_files_${type} "${in_file_path}")
         endforeach()
         if(in_files_${type})
+            list(APPEND out_dirs "${fw_output_header_dir}")
             list(APPEND copy_commands
                 COMMAND ${CMAKE_COMMAND} -E copy ${in_files_${type}} "${fw_output_header_dir}")
             list(APPEND in_files ${in_files_${type}})
-- 
2.51.2

